core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +136,30 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
-            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
-                    stack.getName(), stack.getStatus()));
+    public void validateStackStatusForStartHostGroup(StackDto stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentV4Request) {
+
+        if (stack.getStack().isModificationInProgress()) {
+            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Upscaling is not allowed.",
+                    stack.getStack().getName(), stack.getStack().getStatus()));
+        }
+
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
+        Set<String> dependentComponents = processor.getNonGatewayComponentsByHostGroup().get(instanceGroupAdjustmentV4Request.getInstanceGroup()).stream()
+                .map(srcComponent -> roleDependencyService.getRoleDependency(srcComponent))
+                .flatMap(Collection::stream)
+                .collect(toSet());
+        List<String> unhealthyHostGroupNames = processor.getUnhealthyDependentHostGroups(stack, dependentComponents,
+                UNDEFINED_DEPENDENCY);
+
+        if (!unhealthyHostGroupNames.isEmpty()) {
+            if (unhealthyHostGroupNames.size() > 1 || !unhealthyHostGroupNames.contains(UNDEFINED_DEPENDENCY)) {
+                throw new BadRequestException(format("Upscaling is Not Allowed for HostGroup: '%s' as Data hub '%s' has %d Node(s) Unhealthy.",
```

Comment:
Not sure if this will include the node count. Let's just put in the hostGroup names ... so

```Upscaling is Not Allowed for HostGroup: '%s' as Data hub '%s' as services may not be healthy for instances in  hostGroup(s): [%s]```

Ideally: This would list the instances + state. So if the code that is moved from CmTemplateProcessor to this class were to be simplified to collect the list of hosts, this becomes an easier message to compose.